streamwebs_frontend/streamwebs/tests/models/test_wq_sample.py

```diff
@@ -64,3 +66,44 @@ def test_no_extra_fields(self):
             if not (field.many_to_one and field.related_model is None)
         )))
         self.assertEqual(sorted(fields), sorted(self.expected_fields.keys()))
+
+    # Tests for pH validator.
+    def test_validate_pH_too_large(self):
+        sample_pH_16 = WQ_Sample.objects.create_sample(
+            35, 'Manual', 70,
+            'Manual', 6, 'Manual',
+            16, 'Vernier', 0.879,
+            'Manual', 8.8, 'Vernier',
+            15, 10, 7, 0.93,
+            2.1, 1.9, 14.5, 13
+        )
+
+        with self.assertRaises(ValidationError):
+            validate_pH(sample_pH_16.pH)
+
+    def test_validate_pH_too_small(self):
+        sample_pH_negative_1 = WQ_Sample.objects.create_sample(
+            35, 'Manual', 70,
+            'Manual', 6, 'Manual',
+            -1, 'Vernier', 0.879,
+            'Manual', 8.8, 'Vernier',
+            15, 10, 7, 0.93,
+            2.1, 1.9, 14.5, 13
+        )
+
+        with self.assertRaises(ValidationError):
+            validate_pH(sample_pH_negative_1.pH)
+
+    def test_validate_pH_good(self):
```

Comment:
This test should actually assert something - probably that the return value of validate_pH() is what you would expect for a valid pH. This will error if the code is wrong, but that's not the same as failing a test